presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -71,6 +71,26 @@ public boolean isOutputPartial()
         {
             return outputPartial;
         }
+
+        public static Step partialOutput(Step step)
+        {
+            if (step.isInputRaw()) {
+                return Step.PARTIAL;
+            }
+            else {
+                return Step.INTERMEDIATE;
```

Comment:
Maybe it was a dead code but now it's not anymore. Before spilling `FINAL` (accepting partial input and producing final output) step we have to convert aggregators to `INTERMEDIATE` (accepting partial input and producing partial output).